packages/auth/src/accessToken/service.ts

```diff
@@ -127,31 +139,19 @@ async function createAccessToken(
 
 async function rotate(
   deps: ServiceDependencies,
-  managementId: string,
-  tokenValue: string
+  args: RotateTokenArgs
 ): Promise<AccessToken | undefined> {
-  return AccessToken.transaction(async (trx) => {
-    const oldToken = await AccessToken.query(trx)
-      .delete()
-      .returning('*')
-      .findOne({
-        managementId,
-        value: tokenValue
-      })
-
-    if (!oldToken) {
-      deps.logger.warn(
-        { managementId, tokenValue },
-        'Could not find old token during token rotation'
-      )
-      return
-    }
+  const { managementId, tokenValue, grantId, expiresIn, trx } = args
 
-    return AccessToken.query(trx).insertAndFetch({
-      value: generateToken(),
-      grantId: oldToken.grantId,
-      expiresIn: oldToken.expiresIn,
-      managementId: v4()
-    })
-  })
+  const isRevoked = await revoke(deps, args)
+
+  if (!isRevoked) {
+    deps.logger.warn(
+      { managementId, tokenValue, grantId },
+      'Could not revoke access token'
+    )
+    return undefined
+  }
```

Comment:
Should we stop if we can't revoke the old token?

Reply:
Agreed